pkg/devices_test.go

```diff
@@ -193,3 +194,206 @@ func TestGetCDIForDevice(t *testing.T) {
 		require.Equal(t, test.expectedKind, s.Kind)
 	}
 }
+
+func TestUpdateOCISpecForDevicesWithSpecs(t *testing.T) {
+	testCases := []struct {
+		name           string
+		config         *spec.Spec
+		devices        []string
+		specs          map[string]*cdispec.Spec
+		expectedResult spec.Spec
+		expectedError  bool
+	}{
+		{
+			name:    "add fully qualified device",
+			config:  &spec.Spec{},
+			devices: []string{"vendor.com/device=ABC", "vstartand.org/prodsnow=XYZ"},
+			specs: map[string]*cdispec.Spec{
+				"vendor.com/device": {
+					Version: "0.2.0",
+					Kind:    "vendor.com/device",
+					Devices: []cdispec.Device{
+						{
+							Name: "XYZ",
+							ContainerEdits: cdispec.ContainerEdits{
+								DeviceNodes: []*cdispec.DeviceNode{
+									{
+										Path: "/dev/iceXYZ",
+									},
+								},
+							},
+						},
+						{
+							Name: "ABC",
+							ContainerEdits: cdispec.ContainerEdits{
+								Env: []string{"OR=ICE"},
+								DeviceNodes: []*cdispec.DeviceNode{
+									{
+										Path: "/dev/iceABC",
+									},
+								},
+							},
+						},
+					},
+				},
+				"vstartand.org/prodsnow": {
+					Version: "0.2.0",
+					Kind:    "vstartand.org/prodsnow",
+					Devices: []cdispec.Device{
+						{
+							Name: "XYZ",
+							ContainerEdits: cdispec.ContainerEdits{
+								DeviceNodes: []*cdispec.DeviceNode{
+									{
+										Path: "/dev/snowXYZ",
+									},
+								},
+							},
+						},
+						{
+							Name: "ABC",
+							ContainerEdits: cdispec.ContainerEdits{
+								Env: []string{"AND=SNOW"},
+								DeviceNodes: []*cdispec.DeviceNode{
+									{
+										Path: "/dev/snowABC",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResult: spec.Spec{
+				Process: &spec.Process{
+					Env: []string{"OR=ICE"},
+				},
+				Linux: &spec.Linux{
+					Devices: []spec.LinuxDevice{
```

Comment:
If you had tests that inject more than one device from more than one spec, with more than one of the used specs having Spec-scope `containerEdits` setting environment variables, then it could and should happen. I think currently there are no such tests, so such a false negative is never triggered. If we had those, the fix for correctly handling such cases would be trivial: just replace in the test case verification code
```
require.Equal(t, tc.expectedResult, *tc.config)
```
with
```
require.Equal(t, tc.expectedResult.Linux.Devices, tc.config.Linux.Devices)
require.ElementsMatch(t, tc.expectedResult.Process.Env, tc.config.Process.Env)
```